post/index.md

```diff
@@ -0,0 +1,295 @@
+-   However, navigating the complexities of advanced filtering, especially with nesting and logical operators, can be a daunting task.
+
+Join us as we embark on a hands-on exploration, providing practical examples, insightful strategies, and a roadmap to elevate your Directus proficiency.
+
+Welcome to "Mastering Directus Advanced Filtering" – your comprehensive guide to navigating the depths of Directus with confidence and efficiency.
+
+## Before You Start
+
+> ℹ️ This guide uses the new [Directus SDK 11](https://docs.directus.io/guides/sdk/getting-started.html). Please refer docs if you are using an older version.
+
+### Consider we have Blog Website
+
+-   In our blog website, we have articles list with filters
+
+### Article List
+
+![Article List!](./images/all-articles-list.png)
+
+### Filters
+
+![Article List!](./images/filters.png)
+
+> ## Real World Requirement
+
+-   We need to show articles as per the **filters** are selected ( we have month , year filter ) and **search** and **sort**
+-   Get data Month wise , Year wise , also `when month and year both are selected`
+-   Get Data when search value is entered
+-   when combination of above is happened ( month + year + search )
+-   show Count of data as per the filters
+
+---
+
+### Directus Schema Setup
+
+-   Set up your schema as shown in screenshot
+
+![Blog Schema](./images/blog-schema.png)
+
+-   Now create `published_date` with `date` format
+
+![Article List!](./images/published-date-field.png)
+
+### Accessing Items ( JS Part)
+
+> Items are individual pieces of data in your database. They can be anything, from articles, to IoT status checks. Learn more about [Items](https://docs.directus.io/user-guide/overview/glossary.html#items).
+
+### Initialize Directus instance
+
+```js
+import { createDirectus, rest, aggregate, readItems } from "@directus/sdk";
+
+const client = createDirectus("https://directus.example.com").with(rest());
+```
+
+### Get Data by Month
+
+```js
+  restClient
+        .request(
+            readItems("articles", {
+                filter: {
+                    _or: [
+                        {
+                            "month(published_date)": {
+                                // January
+                                _eq: 1,
+                            },
+                        },
+                        {
+                            "month(published_date)": {
+                                // Feb
+                                _eq: 2,
+                            },
+                        },
+                        {
+                            "month(published_date)": {
+                                // March
+                                _eq: 3,
+                            },
+                        },
+                    ],
+                },
+            })
+        )
+
+// Response
+[{...}] // total length 1
+```
```

Comment:
````suggestion
### Get Data By Month

The following filter will return any item in the `articles` collection that is published in months 1, 2, or 3 (January, February, or March). 

This filter uses the `_or` [filter operator](https://docs.directus.io/reference/filter-rules.html#filter-operators) which contains an array. If at least one of the object conditions are `true`, it will be included.

`month()` is a [DateTime function](https://docs.directus.io/reference/query.html#datetime-functions) that will extract the month from a datetime, date, or timestamp field and return the month number. 

```js
client.request(
      readItems("articles", {
          filter: {
              _or: [
                  {
                      "month(published_date)": {
                          // January
                          _eq: 1,
                      },
                  },
                  {
                      "month(published_date)": {
                          // Feb
                          _eq: 2,
                      },
                  },
                  {
                      "month(published_date)": {
                          // March
                          _eq: 3,
                      },
                  },
              ],
          },
      })
  );
```

Save the file, and run `node index.js` to try it out and see an array of matching items. Of course, you require at least one item which matches the criteria to receive a non-empty array.
````

(I'll fix code indentation when publishing)